packages/miniflare/src/workers/core/proxy.worker.ts

```diff
@@ -255,7 +259,27 @@ export class ProxyServer implements DurableObject {
 			}
 			assert(Array.isArray(args));
 			try {
-				result = func.apply(target, args);
+				if (func.constructor.name === "RpcProperty") {
```

Comment:
Do you need to support `DurableObject` as well?

Reply:
durable objects should already be supported 🤔

Reply:
I meant `class ... extends DurableObject` - these support RPC methods

Reply:
I see... I assume that those will just be wrapped into an `RpcPromise` here and their properties will be 
be constructed as `RpcProperty`, I'll have a look to make sure! 🙂👍 

(PS: I'm taking over the PR as Samuel's OOO and we'd like to land this soon-ish)

Reply:
Yeah I can confirm that in DOs we also get the functions as `RpcProperty`

And that the current implementation works as you'd expect it to 🙂 

![Screenshot 2024-04-13 at 15 34 03](https://github.com/cloudflare/workers-sdk/assets/61631103/6e5953c4-3338-4d0b-9a0c-a231d8ce22df)

![Screenshot 2024-04-13 at 15 34 13](https://github.com/cloudflare/workers-sdk/assets/61631103/86e833c2-7705-4c65-b9e7-d6a09a9ee992)

However I don't think that's going to work currently with `getPlatformProxy` since DO rpc stubs are not supported with multiple `wrangler dev` sessions (which is what `getPlatformProxy` relies on):
![Screenshot 2024-04-12 at 19 29 43](https://github.com/cloudflare/workers-sdk/assets/61631103/d1f351cf-6785-4709-9fbe-c00004b343b6)

We'll have to implement that too, but that feels tangentially related and I'd leave it as a followup not to scope creep this PR 🙂